lib/ui/painting.dart

```diff
@@ -840,6 +840,12 @@ abstract class Image extends NativeFieldWrapperClass2 {
   /// after this method is called.
   void dispose() native "Image_dispose";
 
+  /// For frames of animated images the duration of the current frame in
+  /// milliseconds.
+  ///
+  /// For non animated frames this will be -1.
+  int get durationMillis native "Image_duration";
```

Comment:
Move properties like this over to the object that wraps SkCodec.  There should be a way to obtain the metadata of the animated image without decoding each individual frame.

(In general I'd suggest keeping the Dart wrapper API fairly similar to the underlying Skia API)